pyproject.toml

```
@@ -249,3 +270,59 @@ reportMissingParameterType = false  # 4000+ errors
 reportUnknownVariableType = false  # 6000+ errors
 reportUnknownArgumentType = false  # 6000+ errors
 reportUnknownMemberType = false  # 20000+ errors
+
+[tool.uv]
+default-groups = [
+	"dev",
+	"docs",
+	"lint",
+]
+package = false
+python-preference = "only-managed"
+environments = ["sys_platform == 'win32'"]
+
+[tool.uv.sources]
+configobj = { git = "https://github.com/DiffSK/configobj", rev = "8be54629ee7c26acb5c865b74c76284e80f3aa31" }
+
+[dependency-groups]
+dev = [
+	# NVDA's build system is SCons
+	"SCons==4.8.1",
+	# Packaging NVDA
+	"py2exe==0.13.0.2",
+	# l10nUtil requires Crowdin
+	"crowdin-api-client==1.21.0",
+]
+lint = [
+	"ruff==0.8.1",
+	"pre-commit==4.0.1",
+	"pyright==1.1.396",
+]
+license-check = [
+	"licensecheck==2024.3",
+]
+docs = [
+	# Building user documentation
+	"Markdown==3.7",
+	"mdx_truly_sane_lists==1.3",
+	"markdown-link-attr-modifier==0.2.1",
+	"mdx-gh-links==0.4",
+	# For building developer documentation
+	"sphinx==8.1.2",
+	"sphinx_rtd_theme==3.0.1",
+]
+l10n = [
+	# l10nUtil requires Crowdin
+	"crowdin-api-client==1.21.0"
+]
+system-tests = [
+	"robotframework==7.1.1",
+	"robotremoteserver==1.1.1",
+	"robotframework-screencaplibrary==1.6.0",
+]
+unit-tests = [
+	# Creating XML unit test reports
+	"unittest-xml-reporting==3.2.0",
+	# Feed parameters to tests neatly
+	"parameterized==0.9.0",
+]
```

You define version as dynamic, but without explicit instructions on how to fill that setuptools will set it to 0.0.0 if you remove `package = false`.

```suggestion
]

[tool.setuptools.dynamic]
version = {attr = "buildVersion.version"}
```

The above suggestion works, but uv does not accept a package version in the form `source-xxx`, so this fails to actually sync.